common/params.cc

```diff
@@ -327,3 +334,31 @@ void Params::clearAll(ParamKeyType key_type) {
 
   fsync_dir(getParamPath());
 }
+
+void Params::putNonBlocking(const std::string &key, const std::string &val) {
+  static AsyncWriter async_writer;
```

Comment:
this whole thing seems simpler if the AsyncWriter belongs to the Params instance. I think it's actually desirable that it blocks if you do `Params().putNonBlocking()`. we really shouldn't be repeatedly creating params instances

Reply:
I'll try that